include/dap/session.h

```diff
@@ -251,21 +262,57 @@ class Session {
   virtual bool send(const TypeInfo*, const void* event) = 0;
 };
 
-template <typename F, typename T>
-Session::IsRequest<T> Session::registerHandler(F&& handler) {
-  using ResponseType = typename T::Response;
-  auto cb = [handler](const void* args, const RequestSuccessCallback& onSuccess,
-                      const RequestErrorCallback& onError) {
+template <typename F, typename RequestType>
+Session::IsRequestHandlerWithoutCallback<F> Session::registerHandler(
+    F&& handler) {
+  using ResponseType = typename RequestType::Response;
+  const TypeInfo* typeinfo = TypeOf<RequestType>::type();
+  registerHandler(typeinfo, [handler](const void* args,
```

Comment:
Real shame we're stuck at C++11, as we keep copying functions by value capture. In C++14, I would say to change `[handler]` to `[handler = std::forward<F>(handler)]`.

Reply:
I think this is complicated enough without additional smell from optimizations. If anyone complains about performance, maybe we should just consider upping to a newer C++